lara-typescript/src/section-authoring/util/move-utils.tsx

```diff
@@ -75,16 +80,25 @@ export const moveSection = (args: IMoveSectionSignature) => {
   const realDestSectionIndex = relativeLocation === RelativeLocation.After
     ? destSectionIndex + 1
     : destSectionIndex;
-  const nextSourcePagePageSections = [... sourcePage.sections];
-  const nextDestPagePageSections = [... destPage.sections];
+
   const sourceSection = findSection(pages, sectionId);
   if (sourceSection == null) {
-    error(`can't find source section ${sectionId}`);
-    return false;
+    return error(`can't find source section ${sectionId}`);
   }
-  nextSourcePagePageSections.splice(sectionIndex, 1);
-  nextDestPagePageSections.splice(realDestSectionIndex, 0, sourceSection);
-  setPage({...sourcePage, sections: nextSourcePagePageSections});
-  setPage({...destPage, sections: nextDestPagePageSections});
-  return true;
+  // Remove the sourceSection from the sourcePage
```

Comment:
I see we are using `immer` in other places.  That would make this code array much cleaner.